rule.go

```diff
@@ -482,7 +482,7 @@ func (r *Rule) CVE() string {
 
 // Contents returns all *Content for a rule.
 func (r *Rule) Contents() []*Content {
-	var cs []*Content
+	cs := make([]*Content, 0, len(r.Matchers))
```

Comment:
The reason I didn't do this the first time was because this breaks some usages of the results.
If a client does something like:
```
cs := r.Contents()
if len(cs) > 0 {
	// do stuff
}
```

The clients can no longer rely on the length of the returned list reflecting the number of content matches in the rule, because Matchers includes other types like PCRE, ByteMatcher, etc. I probably should have added a test for this function to highlight this case.

One solution is to perform two passes through, though I'm not sure how that will impact performance:
```
func (m Matcher) isContent() bool {
	if c, ok := m.(*Content); ok {
		return true
	}
}

func (r *Rule) Contents() []*Content {
	for _, m := range r.Matchers {
		if m.isContent() {
			i++
		}
	}
	cs := make([]*Content, 0, i)
	for _, m := range r.Matchers {
		if m.isContent() {
			cs = append(cs, c)
		}
	}
	return cs
}
```


One more option might be to keep track of how many contents we have during the current loop, and then realloc at the end, something like this:
https://stackoverflow.com/questions/16748330/does-go-have-no-real-way-to-shrink-a-slice-is-that-an-issue

Reply:
Well, this PR should not change the `len` only the `cap` (capacity) of the slice

Here is a simple example
Cf https://play.golang.org/p/yK4tg8EUNwC
```
package main

import (
	"fmt"
)

func main() {
	a := make([]int, 0, 16)
	a = append(a, 3)
	fmt.Printf("Hello, playground, %d %d", len(a), cap(a))
}
```
More info can be found here : https://tour.golang.org/moretypes/11 and https://tour.golang.org/moretypes/13

This might end over-allocating memory, but the current algorithm might as well end over allocating memory.

Reply:
Odd, I don't know why I thought this (I swear I did this before and it broke something) oh well, LGTM. =)

Reply:
Maybe you had written
`cs := make([]*Content, len(r.Matchers))`
instead of
`cs := make([]*Content, 0, len(r.Matchers))`

Reply:
I think you're probably correct, IIRC I was running into a nil pointer dereference when doing something like:
```
for _, c := range r.Contents() {
  // use c
}
```

And what you suggest would explain that similar to:
https://play.golang.org/p/50dm48-rIPW